modules/parent-join/src/main/java/org/elasticsearch/join/query/HasChildQueryBuilder.java

```diff
@@ -66,21 +68,23 @@
     /**
      * The default minimum number of children that are required to match for the parent to be considered a match.
      */
-    public static final int DEFAULT_MIN_CHILDREN = 0;
+    public static final int DEFAULT_MIN_CHILDREN = 1;
 
     /**
      * The default value for ignore_unmapped.
      */
     public static final boolean DEFAULT_IGNORE_UNMAPPED = false;
-
+    static final String MIN_CHILDREN_0_DEPRECATION_MESSAGE = "[min_children] 0 will be rejected " +
+        "in 8.0, use 1 and your query results won't change.";
```

Comment:
nit: "... starting with 8.0". 
And maybe rephrase the last part a little more formally, e.g. "Using 1 instead of 0 will return the same result set".

Reply:
this is done, will push when we figure out the above.